2021/day14/day14.java

```diff
@@ -0,0 +1,111 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.io.File;
+
+public class day14 {
+    public static class Rule {
+        String pair;
+        String outcome;
+    }
+
+    public static String readTemplate(String filename) {
+        String template = null;
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filename));
+            template = sc.next();
+        } catch (Exception e) {
+            System.out.println("Something went wrong" + e);
+        } finally {
+            sc.close();
+        }
+        return template;
+    }
+
+        /*
+         * Given the name of a file containing a list of rules for creating
+         * new elements in the form of arrow-delimited strings, e.g. "CH -> B",
+         * read and transform them into a list of `Rule`s
+         */
+    public static ArrayList<Rule> readRules(String filename) {
+        Scanner sc = null;
+        ArrayList<Rule> input = new ArrayList<>();
```

Comment:
This could also be renamed from `input` to `rules`

Reply:
I thought about it too, I am going to rename it